packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -0,0 +1,272 @@
+/**
+ * Copyright 2018-2021 BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import { BlockchainApiConnection } from '@kiltprotocol/chain-helpers'
+import {
+  Deposit,
+  IDidIdentifier,
+  IIdentity,
+  SubmittableExtrinsic,
+} from '@kiltprotocol/types'
+
+import { encodeAddress, signatureVerify } from '@polkadot/util-crypto'
+import type { Option, Struct, u128 } from '@polkadot/types'
+import type {
+  AccountId,
+  BlockNumber,
+  Extrinsic,
+  MultiSignature,
+} from '@polkadot/types/interfaces'
+import type { AnyNumber } from '@polkadot/types/types'
+import type { HexString } from '@polkadot/util/types'
+import { KeypairType } from '@polkadot/util-crypto/types'
+import { BN } from '@polkadot/util'
+import Keyring from '@polkadot/keyring'
+
+// TODO: update with string pattern types once available
+/// A KILT-chain specific address, encoded with the KILT 38 network prefix.
+export type KiltAddress = IIdentity['address']
+/// A chain-agnostic address, which can be encoded using any network prefix.
+export type SubstrateAddress = IIdentity['address']
+
+export type Address = KiltAddress | SubstrateAddress
+
+interface ConnectionRecord extends Struct {
+  did: Address
+  deposit: Deposit
+}
+
+/// Type of signatures to link accounts to DIDs.
+export type SignatureType = MultiSignature['type']
+
+/// Type of a linking payload signing function.
+export type LinkingSignerCallback = (
+  payload: Uint8Array,
```

Comment:
Since the polkadot/extension `signRaw` function expects a `HexString` and returns an object with a signature, which is also a `HexString`, I think it would be better to define it here like this.

Reply:
Something like this? https://github.com/KILTprotocol/sdk-js/pull/514/commits/0b3109da2b3df72208b025a936767dc290766c2d